src/sage/repl/preparse.py

```diff
@@ -1189,6 +1190,15 @@ def preparse_numeric_literals(code, extract=False, quotes="'"):
         sage: preparse_numeric_literals('000042')
         'Integer(42)'
 
+    Check that :issue:`40179` is fixed::
+
+        sage: preparse_numeric_literals("1" * 4300) == f"Integer({"1" * 4300})"
+        True
+        sage: preparse_numeric_literals("1" * 4301) == f"Integer({"1" * 4301})"
+        False
+        sage: preparse_numeric_literals("1" * 4301) == f"Integer('{"1" * 4301}')"
```

Comment:
```suggestion
        sage: preparse_numeric_literals("1" * 4301) == f"Integer('{'1' * 4301}')"
```
I think this will work (the test is a bit different). Please check locally before pushing.